tests/unit/network/rest/utils.js

```diff
@@ -0,0 +1,26 @@
+'use strict'; // eslint-disable-line
+
+const assert = require('assert');
+const constants = require('../../../../lib/constants');
+const { parseURL } = require('../../../../lib/network/rest/utils');
+
+describe('parseURL function', () => {
+    [
+        {
+            inputUrl: `${constants.passthroughFileURL}/test`,
+            expectedKey: 'test',
+        },
+        {
+            inputUrl: `${constants.passthroughFileURL}/test with spaces`,
```

Comment:
Maybe you want the spaces encoded as `%20` or `+` here?

Reply:
`url.parse()` (called in parseURL) already encodes the whitespaces for us (before decoding).
In other words, both encoding and decoding occur in `parseURL`. We could still be provide already encoded urls. I'll add is as an additional test.

Reply:
Done